### PR TITLE
♻️Refactor Property Panel

### LINKED ISSUE #323: PropertyPanel: Grüner Okay-Haken ist überflüssig
**Beschreibe das Problem**

Der grüne Okay-Haken im PropertyPanel ist überflüssig. Wenn meine Eingabe keine Probleme macht sollte, erwarte ich als Benutzer auch nicht zwingend eine Rückmeldung. Als erfahrener Nutzer stört es mich im Zweifelsfall sogar, dass ich die letzten Buchtaben des Namens nicht lesen kann, nur damit ein grüner Haken gemalt werden kann, der keine sinnvolle Zusatzinformation bietet (denn ich gehe ja nicht per se davon aus, dass ich eine Fehler gemacht habe).

image

### LINKED ISSUE #324: PropertyPanel: das „x“ zum Löschen des Werts ist überflüssig
**Beschreibe das Problem**

Im PropertyPanel gibt es ein Icon zum Löschen des Wertes von festen Properties wie "Name" oder "Docs". Dieser sieht anders aus als der "X-Button" zum Löschen dynamischer Properties weiter unten (Stichwort inkonsistentes UI).

Dieser Button nimmt in der Breite Platz für die eingetragenen Werte und bietet dem Nutzer sonst keinen echten Mehrwert.

image

### LINKED ISSUE #325: PropertyPanel: Properties - das „x“ zum Löschen von Properties ist de facto ein „x“
**Beschreibe das Problem**

Im PropertyPanel gibt es einen Button zum Löschen von Properties. Dieser sieht anders aus als das "Kreuz-Icon" zum Löschen fester Werte weiter oben (Stichwort inkonsistentes UI).

Außerdem ist dieser ein sprichwörtliches "x" - das habe ich so noch in keiner ernstzunehmenden Anwendung gesehen.

image

### PR DESCRIPTION
**Pull request by Vyperus**
## What did you change?

This branch:

- Refactors the Code-Style of Property Panel.
- Deletes all green "ok"-symbols.
- Removes all remove "x"-buttons from property-inputs.
- Redesigns the remove buttons for extensions slightly.

## How can others test the changes?

Open a diagram in detail-view.

## Issues

Closes #325 
Closes #324 
Closes #323 

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/modules/property-panel/indextabs/extensions/sections/basics/basics.html [hunk 1]
```diff
@@ -12,13 +12,15 @@
       <table>
         <tr repeat.for="property of properties" index.bind="$index">
           <td class="key-value-column">
-            <input type="text" class="key-value-input" value.bind="newNames[$index]" placeholder="Name" change.delegate="changeName($index)">
+            <input type="text" class="key-value-input" value.bind="_newNames[$index]" placeholder="Name" change.delegate="changeName($index)">
```
Comment (alexanderkasten) [inline src/modules/property-panel/indextabs/extensions/sections/basics/basics.html]: if a variable is used in the view, it should be public. => no underscore

File: src/modules/property-panel/indextabs/extensions/sections/basics/basics.html [hunk 2]
```diff
@@ -12,13 +12,15 @@
...
           </td>
           <td class="key-value-column">
-            <input type="text" class="key-value-input" value.bind="newValues[$index]" placeholder="Value" change.delegate="changeValue($index)">
+            <input type="text" class="key-value-input" value.bind="_newValues[$index]" placeholder="Value" change.delegate="changeValue($index)">
```
Comment (alexanderkasten) [inline src/modules/property-panel/indextabs/extensions/sections/basics/basics.html]: same like above

File: src/modules/property-panel/indextabs/extensions/sections/basics/basics.scss [hunk 3]
```diff
@@ -4,4 +4,19 @@
...
+  margin: 2px 0;
+}
+
+.key-value-delete-button {
+  border: none;
+  background: inherit;
+  padding-top: 2px;
+}
+
+.key-value-delete-button i {
```
Comment (rrrene) [inline src/modules/property-panel/indextabs/extensions/sections/basics/basics.scss]: Hier das ` i` bitte weglassen. Das Icon erbt per se die Farbe.

File: src/modules/property-panel/indextabs/extensions/sections/basics/basics.scss [hunk 4]
```diff
@@ -4,4 +4,19 @@
...
+  color: darkgray;
+}
+
+.key-value-delete-button:hover i {
```
Comment (rrrene) [inline src/modules/property-panel/indextabs/extensions/sections/basics/basics.scss]: dto.

File: src/app.scss [hunk 5]
```diff
@@ -5,7 +5,8 @@ body {
 }
 
 a:focus,
-button:focus {
+button:focus,
+input:focus {
   outline: none;
```
Comment (rrrene) [inline src/app.scss]: Wenn wir wirklich auch die Bootstrap-Styles hiermit überschreiben wollen, muss hier noch ein `!important` hin.

File: src/modules/property-panel/indextabs/extensions/sections/basics/basics.scss [hunk 6]
```diff
@@ -4,4 +4,19 @@
 
 .key-value-input {
   width: 100%;
+  margin: 2px 0;
+}
+
+.key-value-delete-button {
+  border: none;
+  background: inherit;
```
Comment (rrrene) [inline src/modules/property-panel/indextabs/extensions/sections/basics/basics.scss]: Warum nicht `transparent`?

File: src/modules/property-panel/indextabs/general/sections/basics/basics.ts [hunk 7]
```diff
@@ -56,9 +56,9 @@ export class BasicsSection implements ISection {
   }
 
   public detached(): void {
-    if (this.validationError) {
-      this.businessObjInPanel.id = this.previousProcessRefId;
-      this.validationController.validate();
+    if (this._validationError) {
```
Comment (alexanderkasten) [inline src/modules/property-panel/indextabs/general/sections/basics/basics.ts]: you could use early exit here

File: src/modules/property-panel/indextabs/general/sections/escalation-event/escalation-event.ts [hunk 8]
```diff
@@ -19,86 +19,39 @@ export class EscalationEventSection implements ISection {
...
-  private modeler: IBpmnModeler;
-  private generalService: GeneralService;
-
-  private escalationCodeVariable: string;
-  private isBoundaryEvent: boolean = true;
-
   public escalations: Array<IEscalation>;
   public selectedId: string;
   public selectedEscalation: IEscalation;
+  private escalationCodeVariable: string;
```
Comment (alexanderkasten) [inline src/modules/property-panel/indextabs/general/sections/escalation-event/escalation-event.ts]: prefix private variables with an underscore

**Pull request merged**